tests/model_hub_tests/transformation_tests/test_stateful_to_stateless_transformation.py

```diff
@@ -8,21 +8,46 @@
 import models_hub_common.utils as utils
 import pytest
 import os
+import re
 
 def get_read_value_ops(model: ov.Model):
     return [op for op in model.get_ops() if op.get_type_name() == 'ReadValue']
 
-def check_desc_tensors(tensors1, tensors2):
-    # order of tensors may not match, comparing by the total amount and names
-    assert len(tensors1) == len(tensors2)
-    assert set(tuple(t.names) for t in tensors1) == set(tuple(t.names) for t in tensors2)
-    for t1 in tensors1:
-        t2_candidates = [t for t in tensors2 if t1.names & t.names]
-        assert len(t2_candidates) == 1
-        t2 = t2_candidates[0]
-        assert t1.names == t2.names
-        assert t1.get_partial_shape() == t2.get_partial_shape()
-        assert t1.get_element_type() == t2.get_element_type()
+def check_desc_tensors(expected_tensors, tensors):
+    # checks if tensors descriptors are same as expected
+    pattern_input = re.compile(R"input_restored.((past_key_values|present)\.(\d+)\.(key|value))")
```

Comment:
Could you elaborate on why this change is needed? I'm a bit confused, are we backward compatible?

Reply:
Same as [here](https://github.com/openvinotoolkit/openvino/pull/25954#discussion_r1800909046)